tests/ga/test_remoteaccess_handler.py

```diff
@@ -15,162 +15,200 @@
 # Requires Python 2.6+ and Openssl 1.0+
 #
 
-import base64
-import errno as errno
-import glob
-import random
-import string
-import tempfile
-import uuid
-
-from datetime import datetime, timedelta, tzinfo
-from azurelinuxagent.common.protocol import get_protocol_util
-import azurelinuxagent.common.utils.shellutil as shellutil
+from datetime import timedelta
 from azurelinuxagent.common.protocol.wire import *
-from azurelinuxagent.common.osutil import get_osutil
-from azurelinuxagent.common.osutil.default import DefaultOSUtil
-from azurelinuxagent.common.utils.cryptutil import CryptUtil
-from azurelinuxagent.common.exception import CryptError
-from azurelinuxagent.ga.remoteaccess import RemoteAccessHandler, REMOTE_USR_EXPIRATION_FORMAT, MAX_TRY_ATTEMPT
+from azurelinuxagent.ga.remoteaccess import RemoteAccessHandler
 from tests.common.osutil.mock_osutil import MockOSUtil
-from tests.protocol.mockwiredata import WireProtocolData, DATA_FILE
 from tests.tools import *
 
+
+info_messages = []
+error_messages = []
+
+
+def get_user_dictionary(users):
+    user_dictionary = {}
+    for user in users:
+        user_dictionary[user[0]] = user
+    return user_dictionary
+
+
+def log_info(msg_format, *args):
+    info_messages.append(msg_format.format(args))
+
+
+def log_error(msg_format, *args):
+    error_messages.append(msg_format.format(args))
+
+
 class TestRemoteAccessHandler(AgentTestCase):
 
+    def setUp(self):
+        super(TestRemoteAccessHandler, self).setUp()
+        del info_messages[:]
+        del error_messages[:]
+
     # add_user tests
-    @patch('azurelinuxagent.common.utils.cryptutil.CryptUtil.decrypt_secret',
-    return_value="]aPPEv}uNg1FPnl?")
-    def test_add_user(self, _):
+    @patch('azurelinuxagent.common.logger.Logger.info', side_effect=log_info)
+    @patch('azurelinuxagent.common.logger.Logger.error', side_effect=log_error)
+    @patch('azurelinuxagent.common.utils.cryptutil.CryptUtil.decrypt_secret', return_value="]aPPEv}uNg1FPnl?")
+    def test_add_user(self, _1, _2, _3):
         rah = RemoteAccessHandler()
         rah.os_util = MockOSUtil()
         tstpassword = "]aPPEv}uNg1FPnl?"
         tstuser = "foobar"
         expiration_date = datetime.utcnow() + timedelta(days=1)
-        expiration = (expiration_date).strftime("%a, %d %b %Y %H:%M:%S ") + "UTC"
         pwd = tstpassword
         rah.add_user(tstuser, pwd, expiration_date)
-        os_util = rah.os_util
-        os_util.__class__ = MockOSUtil
-        self.assertTrue(tstuser in os_util.all_users, "{0} missing from users".format(tstuser))
-        actual_user = os_util.all_users[tstuser]
-        self.assertEqual(actual_user[7], (expiration_date + timedelta(days=1)).strftime("%Y-%m-%d"))
-        self.assertEqual(actual_user[4], "JIT Account")
+        users = get_user_dictionary(rah.os_util.get_users())
+        self.assertTrue(tstuser in users, "{0} missing from users".format(tstuser))
```

Comment:
For future reference, the agent test tools ensure `unittest.TestCase.assertIn()` (and `assertNotIn()`) are available on all Python versions we support, so this could be written
```
self.assertIn(testuser, users)
```